feat: add a shortcut to copy current block-ref

pengx17 opened the pull request:
Add a new shortcut `Ctrl-Shift-y` (`CMD-SHIFT-y` on Mac) to get the current block ref into the clipboard.

I checked the "copy block ref" shortcut in Roam, which is "CMD+SHIFT+C", but I think it will bring up the browser dev-tools.

Upon copied, the user should be able to see a toast 

![image](https://user-images.githubusercontent.com/584378/114748044-11836600-9d84-11eb-8a80-187f4d5df288.png)

Also updated the onboarding/shortcuts section.

![image](https://user-images.githubusercontent.com/584378/114748133-2d870780-9d84-11eb-9b8a-0faba486e2ab.png)

pengx17 commented:
I am not sure if there is a better shortcut option? 🤯

pengx17 commented on src/main/frontend/dicts.cljs:
```diff
@@ -524,6 +525,7 @@ title: How to take dummy notes?
         :content/cut "Ausschneiden"
         :content/make-todos "{1}s erstellen"
         :content/copy-block-ref "Blockreferenz kopieren"
+        :content/copy-current-block-ref "Aktuelle Blockreferenz kopieren"
```
German and French texts are translated with translate.google.com.

defclass commented on src/main/frontend/handler/editor.cljs:
```diff
@@ -1170,6 +1170,19 @@
    (util/copy-to-clipboard! (tap-clipboard block-id))))
 
+(defn copy-current-block-ref!
+  ([_e]
```
May not add a bracket pair when there is only one function body.

tiensonqin commented:
@pengx17 oh, this one is really needed :) Thank you very much!
@weihua-lu Can we merge this PR and add it to the feat/outliner-core branch too?

pengx17 commented:
@tiensonqin Yah! I do need this shortcut myself, and seeing others in the community has this feature request (source?).

weihua-lu reviewed (commented):
@pengx17 in our core refactor version #1664 , I simplified keyboard copy action with one command:

* when in block selection mode, copy selected blocks
* when in edit mode but no text selected, copy current block ref (here is the bit I borrowed code from this PR)
* when in edit mode with text selected, copy selected text as normal

what do you think?

pengx17 commented on src/main/frontend/handler/editor.cljs:
```diff
@@ -1170,6 +1170,19 @@
    (util/copy-to-clipboard! (tap-clipboard block-id))))
 
+(defn copy-current-block-ref!
+  ([_e]
```
Hi @weihua-lu , this is alreadyed fixed :)

pengx17 commented:
> @pengx17 in our core refactor version #1664 , I simplified keyboard copy action with one command:
> 
> * when in block selection mode, copy selected blocks
> * when in edit mode but no text selected, copy current block ref (here is the bit I borrowed code from this PR)
> * when in edit mode with text selected, copy selected text as normal
> 
> what do you think?

I like your idea! Some feedbacks about the new behavior:

- the behavior may surprise the user for the first time because is not very explicit - but if we have toast notifications, this might not be a big issue 
- some shortcuts involve two sequent actions like `re-index` ("mod+c mod+s"). Not sure if the shortcut collision is a problem.

Also, do you want me to migrate the existing code from https://github.com/logseq/logseq/pull/1664/files to the master branch? In my PR, I introduced a new key-binding, but it is not included in the refactored one. I hope the users are OK with the missing shortcut when using the refactored version.

weihua-lu commented:
- for the block ref copy, yes, the notification is included, so i guess that won't be a surprice
- thanks for the noticing, the conflicting issue has taken care of 
  re-index ("mod+c mod+s") is in one event handler where copy ("mod+c") is in different handler, they're separate so triggering each won't be an issue here.
  
  as said above,
  - if we think it's worth introducing a new cmd for `copy current block ref` only, then we can merge above PR.
  - if we agree on the approach of letting 'mod+c' take care of everything, then i guess we can close this for now and wait for the refacterd version out as for now.

pengx17 commented:
I vote to have a dedicated cmd for `copy current block ref`.

pengx17 commented:
As a second thought, I don't want to have this pr become a stumbling stone for the heated refactoring work. Let's reassess later if we do need a dedicated shortcut wanted by the community.

tiensonqin commented:
@pengx17 it's definitely a stumbling stone :)
I think a dedicated shortcut might be needed for a non-edit mode, because `ctrl-c` usually will copy the block text. 
Also, I agree with you on the behavior might be a bit surprise, feel free to open a thread on the forum, we can make the decision later.

tiensonqin commented:
@pengx17 Sorry I mean not a stumbling stone, my bad :)

pengx17 commented:
> @pengx17 Sorry I mean not a stumbling stone, my bad :)

No problem @tiensonqin. I know stuff is overwhelming on both of you and the Logseq team. Looking forward to the refactored version!
